arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -1773,6 +1703,9 @@ void IResearchDataStore::afterTruncate(TRI_voc_tick_t tick,
       // clear
       state.cookie(key, nullptr);
     }
+  } else if (_recoveryRemoves != nullptr) {  // TODO(MBkkt) should be assert?
+    _recoveryRemoves->clear();
+    _recoveryTrx.Abort();
```

Comment:
Does calling `_recoveryTrx.Abort()` allow to put more data into the transaction after the abort?

Reply:
Yes. It will be new Transaction.
like
```cpp
vector trx1;
... do somethign with trx1
trx1 = {};
... do somethign with trx1 (trx2)
```

Reply:
Simplified IndexWriter::Transaction is just a batch of operations.
They could be committed via Commit and dropped via Abort.